src/isolation.rs

```diff
@@ -0,0 +1,308 @@
+use std::{
+	collections::HashMap,
+	ffi::{CString, OsString},
+	fmt, fs,
+	fs::Permissions,
+	os::unix::{ffi::OsStrExt, fs::PermissionsExt},
+	path::PathBuf,
+};
+
+use tempfile::{Builder, TempDir};
+use uuid::Uuid;
+
+/// Creates a temporary directory.
+pub fn create_temp_dir() -> TempDir {
+	let dir = Builder::new()
+		.permissions(Permissions::from_mode(0o700))
+		.prefix("uhyve-")
+		.suffix(&Uuid::new_v4().to_string())
+		.tempdir()
+		.ok()
+		.unwrap_or_else(|| panic!("The temporary directory could not be created."));
+
+	let dir_permissions = dir.path().metadata().unwrap().permissions();
+	assert!(!dir_permissions.readonly());
+
+	dir
+}
+
+/// Wrapper around a `HashMap` to map guest paths to arbitrary host paths.
+///
+/// See [crate::hypercall::open] to see this in practice.
+pub struct UhyveFileMap {
+	files: HashMap<String, OsString>,
+}
+
+impl UhyveFileMap {
+	/// Creates a UhyveFileMap.
+	///
+	/// * `mappings` - A list of host->guest path mappings with the format "./host_path.txt:guest.txt"
+	pub fn new(mappings: &[String]) -> UhyveFileMap {
+		if !mappings.is_empty() {
```

Comment:
I don't think you need this check. You can `collect` from an empty slice as well.